src/Infrastructure/Persistence/Startup.cs

```diff
@@ -19,26 +20,18 @@ internal static class Startup
 
     internal static IServiceCollection AddPersistence(this IServiceCollection services, IConfiguration config)
     {
-        // TODO: there must be a cleaner way to do IOptions validation...
```

Comment:
I knew adding TODO's to the code would one day pay off :-D
Anyways, thanks again for the PR... this is magnificent work!